### PR TITLE
improve parallel testing

### PR DESCRIPTION
**Pull request by carlosala**
Quite important PR!
Now Zemu is going to (truly) be able to run several tests at the same time.

This bundles:
- Update deps
- Small refactors, specially for `Emulator` class
- Allow Zemu to run several containers at the same time

Todo:
- [x] Adapt included tests to run in parallel
- [x] Test on Linux
- [x] Test on MacOS

**Breaking changes:**
- Remove unused `startDelay`, `pressDelay`, and `x11` variables from main `Zemu` class

:link: [zboto Link](https://app.clickup.com/t/32rp5ap)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -35,38 +40,30 @@ import {
...
-  /**
-   * @deprecated [ZEMU] X11 support is deprecated and not supported anymore
-   */
-  X11 = false
-  custom = ''
   startDelay = DEFAULT_START_DELAY
+  custom = ''
+  model = DEFAULT_MODEL
+  sdk = ''
   startText = 'Ready'
```
Comment (ftheirs) [inline src/index.ts:64]: can we move Ready to a DEFAULT_START_TEXT constant?

File: src/index.ts [hunk 2]
```diff
@@ -343,11 +341,11 @@ export default class Zemu {
 
     while (inputSnapshotBufferHex.data.equals(currentSnapshotBufferHex.data)) {
       const currentTime = new Date()
-      const elapsed: any = currentTime.getTime() - start.getTime()
+      const elapsed = currentTime.getTime() - start.getTime()
       if (elapsed > timeout) {
         throw `Timeout waiting for screen to change (${timeout} ms)`
       }
-      Zemu.delay(500)
+      Zemu.delay()
```
Comment (ftheirs) [inline src/index.ts:348]: is it ok to decrease the delay from 500 to 250ms?

File: src/index.ts [hunk 3]
```diff
@@ -586,44 +574,41 @@ export default class Zemu {
...
       this.log(`Check [${elapsed}ms]`)
       current_events_qty = (await this.getEvents()).length
     }
     this.log(`Screen changed`)
   }
 
-  async waitForText(text: any, timeout = 5000, caseSensitive = false) {
+  async waitForText(text: string, timeout = 60000, caseSensitive = false) {
```
Comment (ftheirs) [inline src/index.ts:596]: check if this is still working with a regex and text

File: src/index.ts (hunk 1 above)
Comment (carlosala) [reply src/index.ts:64]: we should, yes!

File: src/index.ts (hunk 2 above)
Comment (carlosala) [reply src/index.ts:348]: Still testing on all platforms👌🏻

**Review by ftheirs — APPROVED** [review]

**Pull request merged**